### PR TITLE
Check for success files based on configuration flag

### PR DESCRIPTION
**Pull request by moulimukherjee**
Adding a configuration flag, based on which the `pathisGood` method would check for `_SUCCESS` file.

### CODE REVIEW

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala [hunk 1]
```diff
@@ -229,7 +229,13 @@ abstract class FileSource extends SchemedSource with LocalSourceOverride with Hf
    * TODO: consider writing a more in-depth version of this method in [[TimePathedSource]] that looks for
    * TODO: missing days / hours etc.
    */
-  protected def pathIsGood(globPattern: String, conf: Configuration) = FileSource.globHasNonHiddenPaths(globPattern, conf)
+  protected def pathIsGood(globPattern: String, conf: Configuration) = {
+    if (conf.getBoolean("scalding.require_success_file", false)) {
```
Comment (oscar-stripe) [inline scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala:233]: can we add this setting to `Config.scala` as well? `.setRequireSuccessFile(value: Boolean): Config` or something to make it easy also for people to programmatically set the option?

Comment (oscar-stripe) [reply scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala:233]: actually, not. Let's leave it as it is so we don't cause any binary incompatibility by cherry picking this for scalding 0.17.4

Comment (moulimukherjee) [reply scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala:233]: Ack 👍. Adding it.

**Comment by oscar-stripe** [comment]
👍 
Looks good.

If this is green I'll publish a 0.17.4 with this change so we can roll it out with minimal work at Stripe.

File: scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala (hunk 1 above)
Comment (moulimukherjee) [reply scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala:233]: Okay.

**Comment by johnynek** [comment]
@ianoc @pankajroark can I get +1 to make a 0.17.4 that has this PR?

**Pull request merged**